File: Partner/partner/src/main/java/com/weeklycoffee/partner/domain/registration/Registration.java
package com.weeklycoffee.partner.domain.registration;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Registration {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private long memberId;
    private String businessRegistrationNumber;
    private String ceoName;
    private String companyIntroduce;
    private String companyAddress;
    private String companyContact;
    private String companyEmail;
    private String bank;
    private String bankAccount;
    private String registrationDate;
}

File: Partner/partnerclient/api/registration.ts
import axios from "axios";

export interface RegistrationResponse {
  registrationId: number;
  memberId: number;
  companyName: string;
  businessRegistrationNumber: string;
  ceoName: string;
  companyIntroduce: string;
  companyAddress: string;
  companyContact: string;
  companyEmail: string;
  bank: string;
  bankAccount: string;
  registrationDate: string;
}

export interface RegistrationRequest {
  memberId: number;
  companyName: string;
  businessRegistrationNumber: string;
  ceoName: string;
  companyIntroduce: string;
  companyAddress: string;
  companyContact: string;
  companyEmail: string;
  bank: string;
  bankAccount: string;
  registrationDate: string;
}

const registrationApi = {
  add: (registration: RegistrationRequest) =>
    axios.post<RegistrationResponse>(
      `${process.env.NEXT_PUBLIC_API_BASE}/registration`,
      registration
    ),
};

export default registrationApi;
